src/sql/base/browser/ui/chart/chart.component.ts

```diff
@@ -0,0 +1,319 @@
+							data: set.data.map(entry => typeof entry === 'number' ? entry : entry.x),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					result.labels = data.labels;
+					break;
+				}
+			case 'pie':
+			case 'doughnut':
+				{
+					const data = <azdata.PieChartData><unknown>val;
+
+					result.datasets.push({
+						data: data.dataset.map(entry => typeof entry.value === 'number' ? entry.value : entry.value.x),
+						backgroundColor: data.dataset.map(entry => entry.backgroundColor),
+						borderColor: data.dataset.map(entry => entry.borderColor)
+					});
+
+					result.labels = data.dataset.map(val => val.dataLabel);
+					break;
+				}
+			case 'scatter':
+				{
+					const data = <azdata.ScatterplotData><unknown>val;
+
+					for (let set of data.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => [entry.x, entry.y]),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					break;
+				}
+			case 'bubble':
+				{
+					const data = <azdata.BubbleChartData><unknown>val;
+
+					for (let set of data.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => ({ x: entry.x, y: entry.y, r: entry.r })),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					break;
+				}
+			case 'polarArea':
+				{
+					const data = <azdata.PolarAreaChartData><unknown>val;
+
+					result.datasets.push({
+						data: data.dataset.map(entry => typeof entry.value === 'number' ? entry.value : entry.value.x),
+						backgroundColor: data.dataset.map(entry => entry.backgroundColor),
+						borderColor: data.dataset.map(entry => entry.borderColor)
+					});
+
+					result.labels = data.dataset.map(val => val.dataLabel);
+					break;
+				}
+			case 'radar':
+				{
+					const data = <azdata.RadarChartData><unknown>val;
+
+					for (let set of data.datasets) {
+						result.datasets.push({
+							data: set.data.map(entry => typeof entry === 'number' ? entry : entry.x),
+							backgroundColor: set.backgroundColor,
+							borderColor: set.borderColor,
+							label: set.dataLabel
+						});
+					}
+
+					result.labels = data.labels;
+					break;
+				}
+			default:
+				throw new Error(`Unsupported chart type: '${this._type}'`);
+		}
+
+		return result;
+	}
+
+	/**
+	 * Function to draw the chart.
+	 * If the chart is already present, a call to this will simply update the chart with new data values (if any).
+	 * Else a new chart will be created.
+	 */
+	public drawChart() {
+		let canvas = document.getElementById(this.element) as HTMLCanvasElement;
+		this.canvas = canvas;
+
+		if (this.chart) {
+			this.chart.data = this._data;
+			this.chart.update();
+		} else {
+			this.chart = new chartjs.Chart(this.canvas.getContext("2d"), {
+				type: <any>this._type.toString(),
```

Comment:
Could you make a convert function instead to verify we're using the correct type in the future? That decouples us from using their exact names too which is nice.